lm_eval/models/optimum.py

```diff
@@ -0,0 +1,1116 @@
+import copy
+import os
+from pathlib import Path
+from typing import List, Literal, Optional, Tuple, Union
+
+import torch
+import torch.nn.functional as F
+import transformers
+from accelerate import Accelerator, DistributedType, find_executable_batch_size
+from packaging import version
+from peft import PeftModel
+from peft import __version__ as PEFT_VERSION
+from tqdm import tqdm
+from transformers.models.auto.modeling_auto import (
+    MODEL_FOR_CAUSAL_LM_MAPPING_NAMES,
+    MODEL_FOR_SEQ_TO_SEQ_CAUSAL_LM_MAPPING_NAMES,
+)
+
+from lm_eval import utils
+from lm_eval.api.instance import Instance
+from lm_eval.api.model import LM
+from lm_eval.api.registry import register_model
+from lm_eval.utils import Collator, stop_sequences_criteria
+
+import optimum
+from optimum.intel.openvino import OVModelForCausalLM
+from lm_eval.api.registry import register_model
+
+eval_logger = utils.eval_logger
+
+
+
+@register_model("optimum-causal")
```

Comment:
Also, are seq2seq models supported in Intel Optimum? if so, consider adding support and subsequently renaming to just `optimum` rather than `optimum-causal`. If not, there should be a check that we are not trying to run a seq2seq model, that raises a descriptive error message.

Reply:
No, seq2seq models are not supported in Intel Optimum.

Reply:
got it! Then this impl. can match Mamba pretty closely.